patches/minecraft/net/minecraft/world/World.java.patch

```diff
@@ -99,8 +126,12 @@
      }
  
      public boolean func_175667_e(BlockPos p_175667_1_)
-@@ -286,39 +318,58 @@
+@@ -284,41 +321,62 @@
+         }
+         else
          {
++            if (p_180501_2_ instanceof net.minecraftforge.common.property.IExtendedBlockState)
++                p_180501_2_ = ((net.minecraftforge.common.property.IExtendedBlockState) p_180501_2_).getClean();
```

Comment:
this is probably only needed in chunk itself, when its being directly added to the array.